contrib/storage-hive/core/src/test/java/org/apache/drill/exec/hive/TestHiveStorage.java

```diff
@@ -42,6 +43,25 @@ public void hiveReadWithDb() throws Exception {
   }
 
   @Test
+  public void simpleLimitZero() throws Exception {
+    testBuilder()
+        .sqlQuery("select * from hive.kv limit 0")
+        .expectsEmptyResultSet()
+        .baselineColumns("key", "value")
+        .go();
+  }
+
+  @Test
+  public void simpleLimitZeroPlan() throws Exception {
+    final String[] expectedPlan = {
+      ".*Limit.*\n" +
+          ".*Values.*"
+    };
+    final String[] excludedPlan = {};
+    PlanTestBase.testPlanMatchingPatterns("select * from hive.kv limit 0", expectedPlan, excludedPlan);
+  }
+
```

Comment:
Also, you may consider adding a negative test, such as "schemed limit 0", but select clause has a function. 

select substr(char_col, 1, 5) from hive.table limit 0;

In such case, I assume this patch would not do the short cut, and the function return type is not known for now in Drill's planning time.